build tersonflow-lite error

### Issue type

Build/Install

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

commit bce088648427f369e814f28f96db4a335f20a520

### Custom code

Yes

### OS platform and distribution

_No response_

### Mobile device

_No response_

### Python version

_No response_

### Bazel version

_No response_

### GCC/compiler version

_No response_

### CUDA/cuDNN version

_No response_

### GPU model and memory

_No response_

### Current behavior?

cmake -DCMAKE_TOOLCHAIN_FILE=/opt/android-ndk-r27c/build/cmake/android.toolchain.cmake -DANDROID_ABI=arm64-v8a ../tensorflow_src/tensorflow/lite -DCMAKE_BUILD_TYPE=Release -DANDROID_PLATFORM=29 -DTFLITE_ENABLE_GPU=OFF -DTFLITE_ENABLE_RUY=ON -DTFLITE_ENABLE_NNAPI=ON -DTFLITE_ENABLE_XNNPACK=ON -DTFLITE_ENABLE_MMAP=ON -DTFLITE_HOST_TOOLS_DIR=/usr/local/bin/

### Standalone code to reproduce the issue

```shell
git clone https://github.com/tensorflow/tensorflow tensorflow_src
mkdir tflite_build
cd tflite_build
cmake -DCMAKE_TOOLCHAIN_FILE=/opt/android-ndk-r27c/build/cmake/android.toolchain.cmake -DANDROID_ABI=arm64-v8a ../tensorflow_src/tensorflow/lite -DCMAKE_BUILD_TYPE=Release -DANDROID_PLATFORM=29 -DTFLITE_ENABLE_GPU=OFF -DTFLITE_ENABLE_RUY=ON -DTFLITE_ENABLE_NNAPI=ON -DTFLITE_ENABLE_XNNPACK=ON -DTFLITE_ENABLE_MMAP=ON -DTFLITE_HOST_TOOLS_DIR=/usr/local/bin/
make -j6
```

### Relevant log output

```shell
make
[  0%] Built target ruy_profiler_instrumentation
[  1%] Built target xnnpack_weight_cache_schema_cc_fbs
[  1%] Built target flatbuffers
[  1%] Built target eight_bit_int_gemm
[  1%] Built target cpuinfo
[  1%] Built target ruy_wait
[  1%] Built target ruy_cpuinfo
[  1%] Built target ruy_tune
[  1%] Built target ruy_system_aligned_alloc
[  1%] Built target ruy_prepacked_cache
[  1%] Built target ruy_allocator
[  1%] Built target ruy_block_map
[  1%] Built target ruy_blocking_counter
[  1%] Built target ruy_denormal
[  1%] Built target ruy_thread_pool
[  1%] Built target ruy_apply_multiplier
[  1%] Built target ruy_kernel_arm
[  1%] Built target ruy_pack_arm
[  1%] Built target ruy_kernel_avx512
[  1%] Built target ruy_pack_avx512
[  1%] Built target ruy_have_built_path_for_avx512
[  2%] Built target ruy_kernel_avx2_fma
[  2%] Built target ruy_pack_avx2_fma
[  2%] Built target ruy_have_built_path_for_avx2_fma
[  2%] Built target ruy_kernel_avx
[  2%] Built target ruy_pack_avx
[  2%] Built target ruy_have_built_path_for_avx
[  2%] Built target ruy_ctx
[  2%] Built target ruy_context
[  2%] Built target ruy_context_get_ctx
[  2%] Built target ruy_trmul
[  2%] Built target ruy_prepare_packed_matrices
[  2%] Built target ruy_frontend
[  2%] Building CXX object CMakeFiles/xnnpack-delegate.dir/delegates/xnnpack/file_util.cc.o
/home/yafeng.wang/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:52:32: error: out-of-line definition of 'Duplicate' does not match any declaration in 'tflite::xnnpack::FileDescriptor'
   52 | FileDescriptor FileDescriptor::Duplicate(int fd) {
      |                                ^~~~~~~~~
/home/yafeng.wang/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:73:17: error: no type named 'Offset' in 'tflite::xnnpack::FileDescriptor'
   73 | FileDescriptor::Offset FileDescriptorView::GetPos() const {
      | ~~~~~~~~~~~~~~~~^
/home/yafeng.wang/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:73:24: error: use of undeclared identifier 'FileDescriptorView'; did you mean 'FileDescriptor'?
   73 | FileDescriptor::Offset FileDescriptorView::GetPos() const {
      |                        ^~~~~~~~~~~~~~~~~~
      |                        FileDescriptor
/home/yafeng.wang/tflite_build/tensorflow-src/tensorflow/lite/delegates/xnnpack/file_util.h:33:7: note: 'FileDescriptor' declared here
   33 | class FileDescriptor {
      |       ^
/home/yafeng.wang/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:77:17: error: no type named 'Offset' in 'tflite::xnnpack::FileDescriptor'
   77 | FileDescriptor::Offset FileDescriptorView::SetPos(
      | ~~~~~~~~~~~~~~~~^
/home/yafeng.wang/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:77:24: error: use of undeclared identifier 'FileDescriptorView'; did you mean 'FileDescriptor'?
   77 | FileDescriptor::Offset FileDescriptorView::SetPos(
      |                        ^~~~~~~~~~~~~~~~~~
      |                        FileDescriptor
/home/yafeng.wang/tflite_build/tensorflow-src/tensorflow/lite/delegates/xnnpack/file_util.h:33:7: note: 'FileDescriptor' declared here
   33 | class FileDescriptor {
      |       ^
/home/yafeng.wang/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:78:21: error: no type named 'Offset' in 'tflite::xnnpack::FileDescriptor'
   78 |     FileDescriptor::Offset position) const {
      |     ~~~~~~~~~~~~~~~~^
/home/yafeng.wang/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:82:17: error: no type named 'Offset' in 'tflite::xnnpack::FileDescriptor'
   82 | FileDescriptor::Offset FileDescriptorView::SetPosFromEnd(
      | ~~~~~~~~~~~~~~~~^
/home/yafeng.wang/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:82:24: error: use of undeclared identifier 'FileDescriptorView'; did you mean 'FileDescriptor'?
   82 | FileDescriptor::Offset FileDescriptorView::SetPosFromEnd(
      |                        ^~~~~~~~~~~~~~~~~~
      |                        FileDescriptor
/home/yafeng.wang/tflite_build/tensorflow-src/tensorflow/lite/delegates/xnnpack/file_util.h:33:7: note: 'FileDescriptor' declared here
   33 | class FileDescriptor {
      |       ^
/home/yafeng.wang/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:83:28: error: unknown type name 'offset'; did you mean 'off_t'?
   83 |     FileDescriptor::Offset offset) const {
      |                            ^~~~~~
      |                            off_t
/opt/android-ndk-r27c/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/sys/types.h:101:17: note: 'off_t' declared here
  101 | typedef int64_t off_t;
      |                 ^
/home/yafeng.wang/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:83:21: error: no type named 'Offset' in 'tflite::xnnpack::FileDescriptor'
   83 |     FileDescriptor::Offset offset) const {
      |     ~~~~~~~~~~~~~~~~^
/home/yafeng.wang/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:84:25: error: use of undeclared identifier 'offset'
   84 |   return XNN_LSEEK(fd_, offset, SEEK_END);
      |                         ^
/home/yafeng.wang/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:87:17: error: no type named 'Offset' in 'tflite::xnnpack::FileDescriptor'
   87 | FileDescriptor::Offset FileDescriptorView::MovePos(
      | ~~~~~~~~~~~~~~~~^
/home/yafeng.wang/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:87:24: error: use of undeclared identifier 'FileDescriptorView'; did you mean 'FileDescriptor'?
   87 | FileDescriptor::Offset FileDescriptorView::MovePos(
      |                        ^~~~~~~~~~~~~~~~~~
      |                        FileDescriptor
/home/yafeng.wang/tflite_build/tensorflow-src/tensorflow/lite/delegates/xnnpack/file_util.h:33:7: note: 'FileDescriptor' declared here
   33 | class FileDescriptor {
      |       ^
/home/yafeng.wang/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:88:28: error: unknown type name 'offset'; did you mean 'off_t'?
   88 |     FileDescriptor::Offset offset) const {
      |                            ^~~~~~
      |                            off_t
/opt/android-ndk-r27c/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/sys/types.h:101:17: note: 'off_t' declared here
  101 | typedef int64_t off_t;
      |                 ^
/home/yafeng.wang/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:88:21: error: no type named 'Offset' in 'tflite::xnnpack::FileDescriptor'
   88 |     FileDescriptor::Offset offset) const {
      |     ~~~~~~~~~~~~~~~~^
/home/yafeng.wang/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:89:25: error: use of undeclared identifier 'offset'
   89 |   return XNN_LSEEK(fd_, offset, SEEK_CUR);
      |                         ^
/home/yafeng.wang/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:103:6: error: use of undeclared identifier 'FileDescriptorView'; did you mean 'FileDescriptor'?
  103 | bool FileDescriptorView::Read(void* dst, size_t count) const {
      |      ^~~~~~~~~~~~~~~~~~
      |      FileDescriptor
/home/yafeng.wang/tflite_build/tensorflow-src/tensorflow/lite/delegates/xnnpack/file_util.h:33:7: note: 'FileDescriptor' declared here
   33 | class FileDescriptor {
      |       ^
/home/yafeng.wang/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:116:6: error: use of undeclared identifier 'FileDescriptorView'; did you mean 'FileDescriptor'?
  116 | bool FileDescriptorView::Write(const void* src, size_t count) const {
      |      ^~~~~~~~~~~~~~~~~~
      |      FileDescriptor
/home/yafeng.wang/tflite_build/tensorflow-src/tensorflow/lite/delegates/xnnpack/file_util.h:33:7: note: 'FileDescriptor' declared here
   33 | class FileDescriptor {
      |       ^
/home/yafeng.wang/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:129:6: error: use of undeclared identifier 'FileDescriptorView'; did you mean 'FileDescriptor'?
  129 | bool FileDescriptorView::Truncate(size_t size) const {
      |      ^~~~~~~~~~~~~~~~~~
      |      FileDescriptor
/home/yafeng.wang/tflite_build/tensorflow-src/tensorflow/lite/delegates/xnnpack/file_util.h:33:7: note: 'FileDescriptor' declared here
   33 | class FileDescriptor {
      |       ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make[2]: *** [CMakeFiles/xnnpack-delegate.dir/delegates/xnnpack/file_util.cc.o] Error 1
make[1]: *** [CMakeFiles/xnnpack-delegate.dir/all] Error 2
make: *** [all] Error 2
```

check out  539a42d194fa52628c12c71722b9759e8672b06c and change code as https://github.com/tensorflow/tensorflow/pull/102674 still error.
``` shell
cmake --build .
[  0%] Building CXX object _deps/ruy-build/ruy/profiler/CMakeFiles/ruy_profiler_instrumentation.dir/instrumentation.cc.o
[  0%] Linking CXX static library libruy_profiler_instrumentation.a
[  0%] Built target ruy_profiler_instrumentation
[  1%] Generating tensorflow/lite/delegates/xnnpack/weight_cache_schema_generated.h
[  1%] Built target xnnpack_weight_cache_schema_cc_fbs
[  1%] Building CXX object _deps/flatbuffers-build/CMakeFiles/flatbuffers.dir/src/idl_parser.cpp.o
[  1%] Building CXX object _deps/flatbuffers-build/CMakeFiles/flatbuffers.dir/src/idl_gen_text.cpp.o
[  1%] Building CXX object _deps/flatbuffers-build/CMakeFiles/flatbuffers.dir/src/reflection.cpp.o
[  1%] Building CXX object _deps/flatbuffers-build/CMakeFiles/flatbuffers.dir/src/util.cpp.o
[  1%] Linking CXX static library libflatbuffers.a
[  1%] Built target flatbuffers
[  1%] Building CXX object _deps/gemmlowp-build/CMakeFiles/eight_bit_int_gemm.dir/__/__/gemmlowp/eight_bit_int_gemm/eight_bit_int_gemm.cc.o
[  1%] Linking CXX static library libeight_bit_int_gemm.a
[  1%] Built target eight_bit_int_gemm
[  1%] Building C object _deps/cpuinfo-build/CMakeFiles/cpuinfo.dir/src/api.c.o
[  1%] Building C object _deps/cpuinfo-build/CMakeFiles/cpuinfo.dir/src/cache.c.o
[  1%] Building C object _deps/cpuinfo-build/CMakeFiles/cpuinfo.dir/src/init.c.o
[  1%] Building C object _deps/cpuinfo-build/CMakeFiles/cpuinfo.dir/src/log.c.o
[  1%] Building C object _deps/cpuinfo-build/CMakeFiles/cpuinfo.dir/src/arm/uarch.c.o
[  1%] Building C object _deps/cpuinfo-build/CMakeFiles/cpuinfo.dir/src/arm/cache.c.o
[  1%] Building C object _deps/cpuinfo-build/CMakeFiles/cpuinfo.dir/src/arm/linux/init.c.o
[  1%] Building C object _deps/cpuinfo-build/CMakeFiles/cpuinfo.dir/src/arm/linux/cpuinfo.c.o
[  1%] Building C object _deps/cpuinfo-build/CMakeFiles/cpuinfo.dir/src/arm/linux/clusters.c.o
[  1%] Building C object _deps/cpuinfo-build/CMakeFiles/cpuinfo.dir/src/arm/linux/chipset.c.o
[  1%] Building C object _deps/cpuinfo-build/CMakeFiles/cpuinfo.dir/src/arm/linux/midr.c.o
[  1%] Building C object _deps/cpuinfo-build/CMakeFiles/cpuinfo.dir/src/arm/linux/hwcap.c.o
[  1%] Building C object _deps/cpuinfo-build/CMakeFiles/cpuinfo.dir/src/arm/linux/aarch64-isa.c.o
[  1%] Building C object _deps/cpuinfo-build/CMakeFiles/cpuinfo.dir/src/arm/android/properties.c.o
[  1%] Building C object _deps/cpuinfo-build/CMakeFiles/cpuinfo.dir/src/linux/smallfile.c.o
[  1%] Building C object _deps/cpuinfo-build/CMakeFiles/cpuinfo.dir/src/linux/multiline.c.o
[  1%] Building C object _deps/cpuinfo-build/CMakeFiles/cpuinfo.dir/src/linux/cpulist.c.o
[  1%] Building C object _deps/cpuinfo-build/CMakeFiles/cpuinfo.dir/src/linux/processors.c.o
[  1%] Linking C static library libcpuinfo.a
[  1%] Built target cpuinfo
[  1%] Building CXX object _deps/ruy-build/ruy/CMakeFiles/ruy_wait.dir/wait.cc.o
[  1%] Linking CXX static library libruy_wait.a
[  1%] Built target ruy_wait
[  1%] Building CXX object _deps/ruy-build/ruy/CMakeFiles/ruy_cpuinfo.dir/cpuinfo.cc.o
[  1%] Linking CXX static library libruy_cpuinfo.a
[  1%] Built target ruy_cpuinfo
[  1%] Building CXX object _deps/ruy-build/ruy/CMakeFiles/ruy_tune.dir/tune.cc.o
[  1%] Linking CXX static library libruy_tune.a
[  1%] Built target ruy_tune
[  1%] Building CXX object _deps/ruy-build/ruy/CMakeFiles/ruy_system_aligned_alloc.dir/system_aligned_alloc.cc.o
[  1%] Linking CXX static library libruy_system_aligned_alloc.a
[  1%] Built target ruy_system_aligned_alloc
[  1%] Building CXX object _deps/ruy-build/ruy/CMakeFiles/ruy_prepacked_cache.dir/prepacked_cache.cc.o
[  2%] Linking CXX static library libruy_prepacked_cache.a
[  2%] Built target ruy_prepacked_cache
[  3%] Building CXX object _deps/ruy-build/ruy/CMakeFiles/ruy_allocator.dir/allocator.cc.o
[  3%] Linking CXX static library libruy_allocator.a
[  3%] Built target ruy_allocator
[  3%] Building CXX object _deps/ruy-build/ruy/CMakeFiles/ruy_block_map.dir/block_map.cc.o
[  3%] Linking CXX static library libruy_block_map.a
[  3%] Built target ruy_block_map
[  3%] Building CXX object _deps/ruy-build/ruy/CMakeFiles/ruy_blocking_counter.dir/blocking_counter.cc.o
[  3%] Linking CXX static library libruy_blocking_counter.a
[  3%] Built target ruy_blocking_counter
[  3%] Building CXX object _deps/ruy-build/ruy/CMakeFiles/ruy_denormal.dir/denormal.cc.o
[  3%] Linking CXX static library libruy_denormal.a
[  3%] Built target ruy_denormal
[  3%] Building CXX object _deps/ruy-build/ruy/CMakeFiles/ruy_thread_pool.dir/thread_pool.cc.o
[  3%] Linking CXX static library libruy_thread_pool.a
[  3%] Built target ruy_thread_pool
[  3%] Building CXX object _deps/ruy-build/ruy/CMakeFiles/ruy_apply_multiplier.dir/apply_multiplier.cc.o
[  3%] Linking CXX static library libruy_apply_multiplier.a
[  3%] Built target ruy_apply_multiplier
[  3%] Building CXX object _deps/ruy-build/ruy/CMakeFiles/ruy_kernel_arm.dir/kernel_arm32.cc.o
[  3%] Building CXX object _deps/ruy-build/ruy/CMakeFiles/ruy_kernel_arm.dir/kernel_arm64.cc.o
[  3%] Linking CXX static library libruy_kernel_arm.a
[  3%] Built target ruy_kernel_arm
[  3%] Building CXX object _deps/ruy-build/ruy/CMakeFiles/ruy_pack_arm.dir/pack_arm.cc.o
[  3%] Linking CXX static library libruy_pack_arm.a
[  3%] Built target ruy_pack_arm
[  3%] Building CXX object _deps/ruy-build/ruy/CMakeFiles/ruy_kernel_avx512.dir/kernel_avx512.cc.o
[  3%] Linking CXX static library libruy_kernel_avx512.a
[  3%] Built target ruy_kernel_avx512
[  3%] Building CXX object _deps/ruy-build/ruy/CMakeFiles/ruy_pack_avx512.dir/pack_avx512.cc.o
[  3%] Linking CXX static library libruy_pack_avx512.a
[  3%] Built target ruy_pack_avx512
[  3%] Building CXX object _deps/ruy-build/ruy/CMakeFiles/ruy_have_built_path_for_avx512.dir/have_built_path_for_avx512.cc.o
[  3%] Linking CXX static library libruy_have_built_path_for_avx512.a
[  3%] Built target ruy_have_built_path_for_avx512
[  3%] Building CXX object _deps/ruy-build/ruy/CMakeFiles/ruy_kernel_avx2_fma.dir/kernel_avx2_fma.cc.o
[  3%] Linking CXX static library libruy_kernel_avx2_fma.a
[  3%] Built target ruy_kernel_avx2_fma
[  3%] Building CXX object _deps/ruy-build/ruy/CMakeFiles/ruy_pack_avx2_fma.dir/pack_avx2_fma.cc.o
[  3%] Linking CXX static library libruy_pack_avx2_fma.a
[  3%] Built target ruy_pack_avx2_fma
[  3%] Building CXX object _deps/ruy-build/ruy/CMakeFiles/ruy_have_built_path_for_avx2_fma.dir/have_built_path_for_avx2_fma.cc.o
[  3%] Linking CXX static library libruy_have_built_path_for_avx2_fma.a
[  3%] Built target ruy_have_built_path_for_avx2_fma
[  3%] Building CXX object _deps/ruy-build/ruy/CMakeFiles/ruy_kernel_avx.dir/kernel_avx.cc.o
[  3%] Linking CXX static library libruy_kernel_avx.a
[  3%] Built target ruy_kernel_avx
[  3%] Building CXX object _deps/ruy-build/ruy/CMakeFiles/ruy_pack_avx.dir/pack_avx.cc.o
[  3%] Linking CXX static library libruy_pack_avx.a
[  3%] Built target ruy_pack_avx
[  3%] Building CXX object _deps/ruy-build/ruy/CMakeFiles/ruy_have_built_path_for_avx.dir/have_built_path_for_avx.cc.o
[  3%] Linking CXX static library libruy_have_built_path_for_avx.a
[  3%] Built target ruy_have_built_path_for_avx
[  3%] Building CXX object _deps/ruy-build/ruy/CMakeFiles/ruy_ctx.dir/ctx.cc.o
[  3%] Linking CXX static library libruy_ctx.a
[  3%] Built target ruy_ctx
[  3%] Building CXX object _deps/ruy-build/ruy/CMakeFiles/ruy_context.dir/context.cc.o
[  3%] Linking CXX static library libruy_context.a
[  3%] Built target ruy_context
[  3%] Building CXX object _deps/ruy-build/ruy/CMakeFiles/ruy_context_get_ctx.dir/context_get_ctx.cc.o
[  3%] Linking CXX static library libruy_context_get_ctx.a
[  3%] Built target ruy_context_get_ctx
[  3%] Building CXX object _deps/ruy-build/ruy/CMakeFiles/ruy_trmul.dir/trmul.cc.o
[  3%] Linking CXX static library libruy_trmul.a
[  3%] Built target ruy_trmul
[  3%] Building CXX object _deps/ruy-build/ruy/CMakeFiles/ruy_prepare_packed_matrices.dir/prepare_packed_matrices.cc.o
[  3%] Linking CXX static library libruy_prepare_packed_matrices.a
[  3%] Built target ruy_prepare_packed_matrices
[  3%] Building CXX object _deps/ruy-build/ruy/CMakeFiles/ruy_frontend.dir/frontend.cc.o
[  3%] Linking CXX static library libruy_frontend.a
[  3%] Built target ruy_frontend
[  3%] Building CXX object CMakeFiles/xnnpack-delegate.dir/delegates/xnnpack/file_util.cc.o
/mnt/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:52:32: error: out-of-line definition of 'Duplicate' does not match any declaration in 'tflite::xnnpack::FileDescriptor'
   52 | FileDescriptor FileDescriptor::Duplicate(int fd) {
      |                                ^~~~~~~~~
/mnt/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:73:17: error: no type named 'Offset' in 'tflite::xnnpack::FileDescriptor'
   73 | FileDescriptor::Offset FileDescriptorView::GetPos() const {
      | ~~~~~~~~~~~~~~~~^
/mnt/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:73:24: error: use of undeclared identifier 'FileDescriptorView'; did you mean 'FileDescriptor'?
   73 | FileDescriptor::Offset FileDescriptorView::GetPos() const {
      |                        ^~~~~~~~~~~~~~~~~~
      |                        FileDescriptor
/mnt/tflite_build/tensorflow-src/tensorflow/lite/delegates/xnnpack/file_util.h:33:7: note: 'FileDescriptor' declared here
   33 | class FileDescriptor {
      |       ^
/mnt/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:77:17: error: no type named 'Offset' in 'tflite::xnnpack::FileDescriptor'
   77 | FileDescriptor::Offset FileDescriptorView::SetPos(
      | ~~~~~~~~~~~~~~~~^
/mnt/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:77:24: error: use of undeclared identifier 'FileDescriptorView'; did you mean 'FileDescriptor'?
   77 | FileDescriptor::Offset FileDescriptorView::SetPos(
      |                        ^~~~~~~~~~~~~~~~~~
      |                        FileDescriptor
/mnt/tflite_build/tensorflow-src/tensorflow/lite/delegates/xnnpack/file_util.h:33:7: note: 'FileDescriptor' declared here
   33 | class FileDescriptor {
      |       ^
/mnt/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:78:21: error: no type named 'Offset' in 'tflite::xnnpack::FileDescriptor'
   78 |     FileDescriptor::Offset position) const {
      |     ~~~~~~~~~~~~~~~~^
/mnt/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:82:17: error: no type named 'Offset' in 'tflite::xnnpack::FileDescriptor'
   82 | FileDescriptor::Offset FileDescriptorView::SetPosFromEnd(
      | ~~~~~~~~~~~~~~~~^
/mnt/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:82:24: error: use of undeclared identifier 'FileDescriptorView'; did you mean 'FileDescriptor'?
   82 | FileDescriptor::Offset FileDescriptorView::SetPosFromEnd(
      |                        ^~~~~~~~~~~~~~~~~~
      |                        FileDescriptor
/mnt/tflite_build/tensorflow-src/tensorflow/lite/delegates/xnnpack/file_util.h:33:7: note: 'FileDescriptor' declared here
   33 | class FileDescriptor {
      |       ^
/mnt/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:83:28: error: unknown type name 'offset'; did you mean 'off_t'?
   83 |     FileDescriptor::Offset offset) const {
      |                            ^~~~~~
      |                            off_t
/opt/android-ndk-r27d/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/sys/types.h:101:17: note: 'off_t' declared here
  101 | typedef int64_t off_t;
      |                 ^
/mnt/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:83:21: error: no type named 'Offset' in 'tflite::xnnpack::FileDescriptor'
   83 |     FileDescriptor::Offset offset) const {
      |     ~~~~~~~~~~~~~~~~^
/mnt/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:84:25: error: use of undeclared identifier 'offset'
   84 |   return XNN_LSEEK(fd_, offset, SEEK_END);
      |                         ^
/mnt/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:87:17: error: no type named 'Offset' in 'tflite::xnnpack::FileDescriptor'
   87 | FileDescriptor::Offset FileDescriptorView::MovePos(
      | ~~~~~~~~~~~~~~~~^
/mnt/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:87:24: error: use of undeclared identifier 'FileDescriptorView'; did you mean 'FileDescriptor'?
   87 | FileDescriptor::Offset FileDescriptorView::MovePos(
      |                        ^~~~~~~~~~~~~~~~~~
      |                        FileDescriptor
/mnt/tflite_build/tensorflow-src/tensorflow/lite/delegates/xnnpack/file_util.h:33:7: note: 'FileDescriptor' declared here
   33 | class FileDescriptor {
      |       ^
/mnt/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:88:28: error: unknown type name 'offset'; did you mean 'off_t'?
   88 |     FileDescriptor::Offset offset) const {
      |                            ^~~~~~
      |                            off_t
/opt/android-ndk-r27d/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/sys/types.h:101:17: note: 'off_t' declared here
  101 | typedef int64_t off_t;
      |                 ^
/mnt/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:88:21: error: no type named 'Offset' in 'tflite::xnnpack::FileDescriptor'
   88 |     FileDescriptor::Offset offset) const {
      |     ~~~~~~~~~~~~~~~~^
/mnt/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:89:25: error: use of undeclared identifier 'offset'
   89 |   return XNN_LSEEK(fd_, offset, SEEK_CUR);
      |                         ^
/mnt/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:103:6: error: use of undeclared identifier 'FileDescriptorView'; did you mean 'FileDescriptor'?
  103 | bool FileDescriptorView::Read(void* dst, size_t count) const {
      |      ^~~~~~~~~~~~~~~~~~
      |      FileDescriptor
/mnt/tflite_build/tensorflow-src/tensorflow/lite/delegates/xnnpack/file_util.h:33:7: note: 'FileDescriptor' declared here
   33 | class FileDescriptor {
      |       ^
/mnt/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:116:6: error: use of undeclared identifier 'FileDescriptorView'; did you mean 'FileDescriptor'?
  116 | bool FileDescriptorView::Write(const void* src, size_t count) const {
      |      ^~~~~~~~~~~~~~~~~~
      |      FileDescriptor
/mnt/tflite_build/tensorflow-src/tensorflow/lite/delegates/xnnpack/file_util.h:33:7: note: 'FileDescriptor' declared here
   33 | class FileDescriptor {
      |       ^
/mnt/tensorflow_src/tensorflow/lite/delegates/xnnpack/file_util.cc:129:6: error: use of undeclared identifier 'FileDescriptorView'; did you mean 'FileDescriptor'?
  129 | bool FileDescriptorView::Truncate(size_t size) const {
      |      ^~~~~~~~~~~~~~~~~~
      |      FileDescriptor
/mnt/tflite_build/tensorflow-src/tensorflow/lite/delegates/xnnpack/file_util.h:33:7: note: 'FileDescriptor' declared here
   33 | class FileDescriptor {
      |       ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
gmake[2]: *** [CMakeFiles/xnnpack-delegate.dir/build.make:82：CMakeFiles/xnnpack-delegate.dir/delegates/xnnpack/file_util.cc.o] 错误 1
gmake[1]: *** [CMakeFiles/Makefile2:1294：CMakeFiles/xnnpack-delegate.dir/all] 错误 2
gmake: *** [Makefile:136：all] 错误 2

```

--- a/tensorflow/lite/delegates/xnnpack/file_util.h
+++ b/tensorflow/lite/delegates/xnnpack/file_util.h
@@ -15,7 +15,9 @@ limitations under the License.
 #ifndef TENSORFLOW_LITE_DELEGATES_XNNPACK_FILE_UTIL_H_
 #define TENSORFLOW_LITE_DELEGATES_XNNPACK_FILE_UTIL_H_
 
+#if !defined(_WIN32)
 #include <sys/types.h>
+#endif
 
 #include <cstddef>
 #include <utility>

